fibonacci/scala/code.scala

```diff
@@ -1,16 +1,24 @@
+//> using scala 3.6.1
+
 package jvm
 
+import scala.annotation.tailrec
+
 @main
 def main(number: String): Unit =
   val u = number.toInt
   var r = 0
-  for i <- 1 until u do
-    r += fibonacci(i)
+  var i = 0
+  while i < u do
+    r += fibonacci(i, 0, 1)
+    i += 1
   println(r)
 
+@tailrec
+def fibonacci(n: Int, a: Int, b: Int): Int =
+  if (n > 0)
+    fibonacci(n - 1, b, a + b)
+  else
+    a
```

Comment:
As far as I understand it, the point of this benchmark is to test the naive non-tail recursive implementation. Every language would be faster if it used the linear time algorithm.

Reply:
seems true, let's me revert it